From 0000000000000000000000000000000000000003 Mon Sep 17 00:00:00 2001
From: Omer Faruk Nehir <omerfaruknehir@gmail.com>
Date: Wed, 5 Aug 2026 06:02:00 +0300
Subject: [PATCH 3/4] arm64: dts: qcom: hamoa: add CAMSS CPAS top resource

The X1E80100 camera-platform resource at 0x0ac19000 is described by the
firmware CAMP device and is used for the AP/AON camera ownership mux.  Add the
window to CAMSS so the media driver can perform a powered, clocked handoff.

The resource matches the Windows/ACPI range 0x0ac19000 + 0x0c000.  The mux
register used by the A14 is at offset 0x1e0 inside this window.

Signed-off-by: Omer Faruk Nehir <omerfaruknehir@gmail.com>
---
 arch/arm64/boot/dts/qcom/hamoa.dtsi | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/hamoa.dtsi b/arch/arm64/boot/dts/qcom/hamoa.dtsi
index 000000000000..000000000001 100644
--- a/arch/arm64/boot/dts/qcom/hamoa.dtsi
+++ b/arch/arm64/boot/dts/qcom/hamoa.dtsi
@@ -1,3 +1,4 @@
+/* This hunk is expected in Ubuntu's X1E camera-enabled hamoa.dtsi. */
@@ -0000,8 +0001,9 @@ camss: isp@acb7000 {
 		      <0 0x0ac62000 0 0x4000>,
 		      <0 0x0ac71000 0 0x4000>,
 		      <0 0x0acc7000 0 0x2000>,
-		      <0 0x0accb000 0 0x2000>;
+		      <0 0x0accb000 0 0x2000>,
+		      <0 0x0ac19000 0 0xc000>;
 
 	reg-names = "csid0", "csid1", "csid2",
 		    "csid_lite0", "csid_lite1", "csid_wrapper",
@@ -0010,7 +0012,7 @@ camss: isp@acb7000 {
 		    "csitpg0", "csitpg1", "csitpg2",
 		    "vfe0", "vfe1",
-		    "vfe_lite0", "vfe_lite1";
+		    "vfe_lite0", "vfe_lite1", "cpas-top";
 
 	clocks = <&camcc CAM_CC_CAMNOC_AXI_NRT_CLK>,
-- 
2.51.0
